sources/controllers/node_controller.rb

```diff
@@ -11,28 +9,35 @@ class NodeController < Sinatra::Base
   end
 
   # html view, sends stuff to /node as POST
-  get '/new' do
-    @node = Node.new
-    haml :node
-  end
-
-  post '/' do
-    Node.new(params[:node])
-  end
+  get '/new' do             #
+    @node = Node.new        # dafuq is this
+    haml :node              #
+  end                       #
 
-  delete '/:id' do
-    Node.delete(params[:id])
-  end
+  post '/' do               #
+    Node.new(params[:node]) # dafuq is that
```

Comment:
get '/new' do             #
      @node = Node.new        # dafuq is this
      haml :node              #
    end                       #

    post '/' do               #
      Node.new(params[:node]) # dafuq is that
    end 
My point is, that those two routes should not be there. The correct way would be the  POST method since you want to create a new node. You should choose one of the two methods on put its content into the POST route.